content/articles/how-to-post-form-data-as-json-using-the-formdata-web-api-and-fetch-api/index.md

```diff
@@ -0,0 +1,362 @@
+  //If the response was not OK, throw an error.
+  if (!res.ok) {
+    let error = await res.text();
+    throw new Error(error);
+  }
+  //If the response was OK, return the response body.
+  return res.json();
+}
+```
+
+### Full script code
+
+This is the full js script with inline comments from the steps above:
+
+1. Capture the form fields using the browser FormData API
+2. Convert the form field data to object and then to JSON.
+3. Finally send them to an API endpoint using the browser Fetch API.
+
+```js
+//Get the form element by id
+let sampleForm = document.getElementById("sample-form");
+
+//Add an event listener to the form element
+sampleForm.addEventListener("submit", handleFormSubmit);
+
+ //Event handler for the form submit an event.
+async function handleFormSubmit(e) {
+  /**
+   * Prevent the default browser submitting behaviour 
+   * so that we can handle it ourselve instead.
+   */
+  e.preventDefault();
+
+  //Get the element attached to the event handler.
+  let form = e.currentTarget;
+
+  //Take the URL from the form's `action` attribute
+  let url = form.action;
+
+  try {
+    /**
+     * Take all the form fields and make the field values
+     * available through a `FormData` instance.
+     */
+    let formFields = new FormData(form);
+
+    //We'll define the `postFormFieldsAsJson()` function in the next step.
+    let responseData = await postFormFieldsAsJson({ url, formFields });
+  } catch (error) {
+
+    // Handle the error here.
+    console.error(`An error has occured ${error}`);
+  }
+}
+
+ //Helper function to POST data as JSON with Fetch.
+async function postFormFieldsAsJson({ url, formData }) {
+
+  //Create an object from the form data entries
+  let formDataAsObject = Object.fromEntries(formData.entries());
+
+  // Format the plain form data as JSON
+  let formDataToJsonString = JSON.stringify(formDataAsObject);
+
+  let fetchOptions = {
+    
+    //HTTP method set to POST.
+    method: "POST",
+    //Set the headers that specify we're sending a JSON body request and accepting JSON response
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+    },
+
+    // POST request body as JSON string.
+    body: formDataToJsonString,
+  };
+
+  //Send the request to the API with the above-defined options.
+  let res = await fetch(url, fetchOptions);
+
+  /**
+  *Get the response body as JSON. 
+  *If the response was not OK, throw an error.
+  */
+  if (!res.ok) {
+    let error = await res.text();
+    throw new Error(error);
+  }
+
+  //If the response was OK, return the response body.
+  return res.json();
+}
+```
+
+> Link to code on [Github](https://github.com/gisioraelvis/sending-formdata-as-json-using-fetch)
+
+The sample form data (`formDataToJsonString`) object sent as JSON;
+
+```json
+{
+  "title": "Male",
+  "username": "gisioraelvis",
+  "usermail": "gisioraelvis@gmail.com"
+}
+```
+
+### Bonus: Handling JSON request body in a Nodejs and Express.js API
```

Comment:
I don't think this should be called bonus, if it's important to the reader, then it should be done and if not, it shouldn't be included.